tango_simlib/tests/test_dish.py

```diff
@@ -0,0 +1,187 @@
+            self.model.sim_actions[cmd_name](data_input=timestamp)
+            num_method_calls = mock_set_val.call_count
+            self.assertEquals(num_method_calls, 2)
+            self.assertEquals(calls, mock_set_val.mock_calls)
+            mock_set_val.reset_mock()
+
+    def test_low_power(self):
+        power_state_quant = self.model.sim_quantities['powerState']
+        power_state_enum_labels = power_state_quant.meta['enum_labels']
+
+        # Write a value to the quantity to override the default one which is a boolean
+        # True value. Pick any value except for 'LOW'.
+        power_state_quant.last_val = power_state_enum_labels.index('OFF')
+
+        dish_mode_quant = self.model.sim_quantities['dishMode']
+        dish_mode_enum_labels = dish_mode_quant.meta['enum_labels']
+
+        for allowed_mode in ['STOW', 'MAINTENANCE']:
+            dish_mode_quant.last_val = dish_mode_enum_labels.index(allowed_mode)
+            self.model.sim_actions['LowPower']()
+            self.assertEqual(power_state_quant.last_val, power_state_enum_labels.index('LOW'))
+
+        for not_allowed_mode in ['OFF', 'STARTUP', 'SHUTDOWN', 'STANDBY-LP',
+                                 'STANDBY-FP', 'CONFIG', 'OPERATE']:
+            dish_mode_quant.last_val = dish_mode_enum_labels.index(not_allowed_mode)
+            self.assertRaises(DevFailed, self.model.sim_actions['LowPower'])
+
+    def test_set_maintenance_mode(self):
+        dish_mode_quant = self.model.sim_quantities['dishMode']
+        dish_mode_enum_labels = dish_mode_quant.meta['enum_labels']
+
+        # Write a value to the quantity to override the default one which is a boolean
+        # True value. Pick any value except for 'MAINTENACE'.
+        dish_mode_quant.last_val = dish_mode_enum_labels.index('STANDBY-LP')
+
+        for allowed_mode in ['STANDBY-LP', 'STANDBY-FP']:
+            dish_mode_quant.last_val = dish_mode_enum_labels.index(allowed_mode)
+            self.model.sim_actions['SetMaintenanceMode']()
+            self.assertEqual(dish_mode_quant.last_val, dish_mode_enum_labels.index('MAINTENANCE'))
+
+        for not_allowed_mode in ['OFF', 'STARTUP', 'SHUTDOWN', 'STOW',
+                                 'MAINTENANCE', 'CONFIG', 'OPERATE']:
+            dish_mode_quant.last_val = dish_mode_enum_labels.index(not_allowed_mode)
+            self.assertRaises(DevFailed, self.model.sim_actions['SetMaintenanceMode'])
+
+    def test_set_operate_mode(self):
+        dish_mode_quant = self.model.sim_quantities['dishMode']
+        dish_mode_enum_labels = dish_mode_quant.meta['enum_labels']
+
+        # Write a value to the quantity to override the default one which is a boolean
+        # True value. Pick any value except for 'MAINTENACE'.
+        dish_mode_quant.last_val = dish_mode_enum_labels.index('STANDBY-LP')
+
+        for allowed_mode in ['STANDBY-FP']:
+            dish_mode_quant.last_val = dish_mode_enum_labels.index(allowed_mode)
```

Comment:
- [ ] Either don't use for, or make the loop correct for future use by resetting at the start.